exec/java-exec/src/test/java/org/apache/drill/exec/coord/zk/TestZKACL.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.curator.test.TestingServer;
+import org.apache.drill.common.config.DrillConfig;
+import org.apache.drill.common.scanner.ClassPathScanner;
+import org.apache.drill.common.scanner.persistence.ScanResult;
+import org.apache.drill.exec.ExecConstants;
+import org.apache.drill.exec.server.BootStrapContext;
+import org.apache.drill.exec.server.options.SystemOptionManager;
+import org.apache.zookeeper.data.ACL;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.List;
+
+public class TestZKACL {
+
+  private TestingServer server;
+  private final static String cluster_config_znode = "test-cluster_config_znode";
+  private final static byte[] cluster_config_data = "drill-node-1".getBytes();
+  private final static String drill_zk_root = "drill-test-drill_zk_root";
+  private final static String drill_cluster_name = "test-drillbits";
+  private static final String drillClusterPath = "/" + drill_zk_root + "/" + drill_cluster_name;
+  private static final RetryPolicy retryPolicy = new RetryNTimes(1, 1000);
+
+  private static final String drillUDFName = "test-udfs";
+  private final static byte[] udf_data = "test-udf-1".getBytes();
+  private static final String drillUDFPath = "/" + drill_zk_root + "/" + drillUDFName;
+  private static  ACLProvider aclProviderDelegate;
+
+  private static CuratorFramework client;
+
+  @Before
+  public void setUp() throws Exception {
+    System.setProperty("zookeeper.authProvider.1", "org.apache.zookeeper.server.auth.SASLAuthenticationProvider");
+    String configPath = (ClassLoader.getSystemResource("zkacltest.conf")).getPath();
+    System.setProperty("java.security.auth.login.config", configPath);
+    server = new TestingServer();
+
+
+    final DrillConfig config = new DrillConfig(DrillConfig.create().withValue(ExecConstants.ZK_ACL_PROVIDER,
+            ConfigValueFactory.fromAnyRef("creator-all")
+    ).withValue(ExecConstants.ZK_APPLY_SECURE_ACL, ConfigValueFactory.fromAnyRef(true)));
+
+    final ScanResult result = ClassPathScanner.fromPrescan(config);
+    final BootStrapContext bootStrapContext =
+            new BootStrapContext(config, SystemOptionManager.createDefaultOptionDefinitions(), result);
+    aclProviderDelegate = ZKACLProviderFactory.getACLProvider(config, drillClusterPath, bootStrapContext);
+
+    server.start();
+
+    client =  CuratorFrameworkFactory.builder().
+            retryPolicy(retryPolicy).
+            connectString(server.getConnectString()).
+            aclProvider(aclProviderDelegate).
+            build();
+    client.start();
+  }
+
+  /**
+   * Test ACLs on znodes required to discover the cluster
+   */
+  // ZK libraries only supports one client instance per-machine per-server and it is cached.
+  // This test will fail when run after other ZK tests that setup the client in a way that will cause this test to fail.
+  @Ignore
+  @Test
```

Comment:
Please _Ignore_ at class level. Also provide category for this test as _SecurityTest.class_. It would be great if you can open a JIRA listing the reason why this tests are ignored and refer that JIRA# in the Ignore tag. Please see [here](https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/rpc/user/security/TestUserBitKerberos.java#L48)

Reply:
Done.